core_java/grep/README.md

```diff
@@ -0,0 +1,39 @@
+# Introduction
+Java grep is an implementation of grep in the Java language. The purpose of this project is to develop core java knowledge.  JavaGrepImp uses java.util.regex to match a given pattern, java.io for file io and slf4j logger.  JavaGrepLambdaImp extends JavaGrepImp to use streams for listing files and reading lines rather than the for loops originally implemented. The project is managed with Maven and has been deployed with docker. 
```

Comment:
Java grep is an implementation of grep in the Java language. The purpose of this project is to develop core java knowledge.  JavaGrepImp uses `java.util.regex` to match a given pattern, `java.io` for file io and `slf4j` logger.  JavaGrepLambdaImp extends `JavaGrepImp` to use streams for listing files and reading lines rather than the for loops originally implemented. The project is managed with Maven and has been deployed with docker.